src/app/features/tech-record/components/tech-record-summary-changes/tech-record-summary-changes.component.html

```diff
@@ -0,0 +1,97 @@
+<ng-container *ngIf="techRecordEdited && techRecordChanges && vehicleType">
+  <app-banner>
+    <ng-container title>Important</ng-container>
+    <ng-container content>Check your changes before submitting the technical record</ng-container>
+  </app-banner>
+
+  <app-icon [icon]="techRecordEdited.techRecord_vehicleType || ''"></app-icon>
+  <app-dynamic-form-group [data]="techRecordEdited" [template]="vehicleSummary"></app-dynamic-form-group>
+
+  <app-accordion-control [class]="'padding'" [sectionState]="sectionState">
+    <ng-container *ngFor="let section of customVehicleTemplate; let i = index">
+      <app-accordion [id]="section.name" [title]="section.label" [isExpanded]="true">
+        <ng-template [ngTemplateOutlet]="accordion" [ngTemplateOutletContext]="{ sectionName: section.name, section: section }"></ng-template>
+      </app-accordion>
+    </ng-container>
+  </app-accordion-control>
+
+  <ng-template #accordion let-sectionName="sectionName" let-section="section">
+    <ng-container [ngSwitch]="sectionName">
+      <app-body *ngSwitchCase="'bodySection'" [techRecord]="techRecordEdited" [disableLoadOptions]="true"></app-body>
+
+      <ng-container *ngSwitchCase="'dimensionsSection'">
+        <app-dimensions
+          *ngIf="
```

Comment:
you could probs bust this out to a function like checkVehicleType() since the checks are all the same, save repeating yourself

Reply:
unfortunately due to the way typescript works out the union types, by doing this the reference to the tech record directly below will no longer resolve, this is unfortunately the only way we can do this